Pokedex/SearchViewController.swift

```diff
@@ -0,0 +1,81 @@
+
+class SearchViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource {
+
+    @IBOutlet weak var topNav: UINavigationItem!
+    let imageView = UIImageView(image: #imageLiteral(resourceName: "Pokedex"))
+    
+    @IBOutlet weak var typesCollectionView: UICollectionView!
+
+    @IBOutlet weak var randomizer: UIButton!
+
+    // Actions
+    @IBOutlet weak var hpTextField: UITextField!
+    @IBOutlet weak var attackTextField: UITextField!
+    @IBOutlet weak var defenseTextField: UITextField!
+    
+    var pokemonArray = PokemonGenerator.getPokemonArray()
+
+    
+    var fontElements: [UIColor] = [UIColor.white, UIColor.white, UIColor.white, UIColor.black, UIColor.white, UIColor.white, UIColor.white, UIColor.black, UIColor.white, UIColor.white, UIColor.black, UIColor.white, UIColor.white, UIColor.white, UIColor.white, UIColor.white, UIColor.white, UIColor.white]
+    var imageElements: [UIImage] = [#imageLiteral(resourceName: "bug"), #imageLiteral(resourceName: "dark"), #imageLiteral(resourceName: "dragon"), #imageLiteral(resourceName: "electric"), #imageLiteral(resourceName: "fairy"), #imageLiteral(resourceName: "fighting"), #imageLiteral(resourceName: "fire"), #imageLiteral(resourceName: "flying"), #imageLiteral(resourceName: "ghost"), #imageLiteral(resourceName: "grass"), #imageLiteral(resourceName: "ground"), #imageLiteral(resourceName: "ice"), #imageLiteral(resourceName: "normal"), #imageLiteral(resourceName: "poison"), #imageLiteral(resourceName: "psychic"), #imageLiteral(resourceName: "rock"), #imageLiteral(resourceName: "steel"), #imageLiteral(resourceName: "water")]
+    var diffElements: [String] = ["Bug", "Dark", "Dragon", "Electric", "Fairy", "Fighting", "Fire", "Flying", "Ghost", "Grass", "Ground", "Ice", "Normal", "Poison", "Psychic", "Rock", "Steel", "Water"]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        typesCollectionView.delegate = self
+        typesCollectionView.dataSource = self
+        
+        randomizer.setBackgroundImage(#imageLiteral(resourceName: "Randomizeme"), for: .normal)
+        topNav.titleView = imageView
+        
+    }
+    
+    //number of rows
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return diffElements.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let eachCell = collectionView.dequeueReusableCell(withReuseIdentifier: "eachButton", for: indexPath) as! TypesButtonCollectionViewCell
+        //let randomTwenty = arc4random_uniform(UInt32(pokemonArray.count))
+        eachCell.backgroundColor = UIColor.blue
+        eachCell.element.setBackgroundImage(imageElements[indexPath.item], for: .normal)
+        eachCell.element.setTitle(diffElements[indexPath.item], for: .normal)
+        eachCell.element.setTitleColor(fontElements[indexPath.item], for: .normal)
```

Comment:
It's hard to see if a user clicks a button like "fire" and the app registers this click. To show the click happened, you usually want to change the cell's background color opacity when you click the button or toggle it. Then change it back to regular opacity when untangled.